File: src/main/java/com/cfgglobal/test/web/api/BranchController.java
package com.cfgglobal.test.web.api;

import com.cfgglobal.test.base.JpaBeanUtil;
import com.cfgglobal.test.domain.Branch;
import com.cfgglobal.test.service.BranchService;
import com.cfgglobal.test.web.base.BaseController;
import io.vavr.collection.HashMap;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletRequest;

@RestController
@RequestMapping(value = "/v1/branch")
@Slf4j
public class BranchController extends BaseController {


    @Autowired
    private BranchService branchService;

    @GetMapping
    public ResponseEntity page(Pageable pageable, HttpServletRequest request) {
        Page<Branch> page = branchService.findBySecurity(request.getMethod(), request.getRequestURI(), HashMap.ofAll(request.getParameterMap()), pageable);
        return ResponseEntity.ok(page);


    }

    @GetMapping("{id}")
    public ResponseEntity<Branch> get(@PathVariable Long id) {
        return ResponseEntity.ok(branchService.findOne(id));
    }

    @PostMapping
    public ResponseEntity<Branch> save(@RequestBody Branch branch) {
        return ResponseEntity.ok(branchService.save(branch));

    }


    @PutMapping("{id}")
    public ResponseEntity save(@PathVariable Long id, @RequestBody Branch branch) {
        Branch oldBranch = branchService.findOne(id);
        JpaBeanUtil.copyNonNullProperties(branch, oldBranch);
        return ResponseEntity.ok(branchService.save(oldBranch));
    }


    @DeleteMapping("delete")
    public ResponseEntity delete(@PathVariable Long id, HttpServletRequest request) {
        branchService.deleteBySecurity(id, request.getMethod(), request.getRequestURI());
        return ResponseEntity.noContent().build();
    }


}

File: src/main/kotlin/com/cfgglobal/test/web/api/BranchController.kt
package com.cfgglobal.test.web.api

import com.cfgglobal.test.domain.Branch
import com.cfgglobal.test.service.BranchService
import com.cfgglobal.test.web.base.BaseController
import com.github.leon.bean.JpaBeanUtil
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import javax.servlet.http.HttpServletRequest

@RestController
@RequestMapping(value = ["/v1/branch"])

class BranchController(
        @Autowired
         val branchService: BranchService

) : BaseController() {

   
    @GetMapping
    fun page(pageable: Pageable, request: HttpServletRequest): ResponseEntity<*> {
        val page = branchService.findBySecurity(request.method, request.requestURI, request.parameterMap, pageable)
        return ResponseEntity.ok(page)
    }

    @GetMapping("{id}")
    operator fun get(@PathVariable id: Long): ResponseEntity<Branch> {
        return ResponseEntity.ok(branchService.findOne(id).get())
    }

    @PostMapping
    fun save(@RequestBody branch: Branch): ResponseEntity<Branch> {
        return ResponseEntity.ok(branchService.save(branch))
    }


    @PutMapping("{id}")
    fun save(@PathVariable id: Long, @RequestBody branch: Branch): ResponseEntity<*> {
        val oldBranch = branchService.findOne(id).get()
        JpaBeanUtil.copyNonNullProperties(branch, oldBranch)
        return ResponseEntity.ok(branchService.save(oldBranch))
    }

    @DeleteMapping("delete")
    fun delete(@PathVariable id: Long, request: HttpServletRequest): ResponseEntity<*> {
        branchService!!.deleteBySecurity(id, request.method, request.requestURI)
        return ResponseEntity.noContent().build<Any>()
    }

}